pylmod/membership.py

```diff
@@ -18,7 +19,89 @@ def __init__(
             self,
             cert,
             urlbase='https://learning-modules.mit.edu:8443/',
+            cuuid=None
     ):
         super(Membership, self).__init__(cert, urlbase)
         # Add service base
         self.urlbase += 'service/membership/'
+        self.course_id = None
+        if cuuid is not None:
+            self.course_id = self.get_course_id(cuuid)
+
+    def get_course_id(self, course_uuid):
+        """Get course id based on uuid.
+
+        Args:
+            uuid (str): course uuid, i.e. /project/mitxdemosite
+
+        Raises:
+            PyLmodUnexpectedData: No course data was returned.
+
+        Returns:
+            int: numeric course id
+
+        """
+        course_data = self.get(
+            'courseguide/course?uuid={uuid}'.format(
+                uuid=course_uuid or self.course_id
+            ),
+            params=None
+        )
+        if 'response' not in course_data:
+            failure_messsage = ('Error in get_course_id '
+                                'for {0} - no response'.format(course_data))
+            log.error(failure_messsage)
+            raise PyLmodUnexpectedData(failure_messsage)
+
+        return course_data['response']['docs'][0]['id']
```

Comment:
This is heavily blind indexing, I think you probably want to catch the exception here and raise a PyLmod exception for the entire object address.